runtime/android/core/src/org/xwalk/core/XWalkActivity.java

```diff
@@ -5,16 +5,560 @@
 package org.xwalk.core;
 
 import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.DownloadManager;
+import android.app.DownloadManager.Request;
+import android.app.DownloadManager.Query;
+import android.app.ProgressDialog;
+import android.content.ActivityNotFoundException;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.DialogInterface.OnClickListener;
+import android.content.DialogInterface.OnDismissListener;
+import android.content.DialogInterface.OnShowListener;
+import android.content.Intent;
+import android.content.IntentFilter;
 import android.content.res.Resources;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.os.Environment;
+import android.util.Log;
+
+import java.lang.Thread;
+import java.util.LinkedList;
+
+import junit.framework.Assert;
+
+import org.xwalk.core.XWalkLibraryListener.LibraryStatus;
 
 /**
  * XWalkActiviy is to support cross package resource loading.
  * It provides method to allow overriding getResources() behavior.
  */
-public abstract class XWalkActivity extends Activity {
+public abstract class XWalkActivity extends Activity implements XWalkLibraryListener {
+    private static final String XWALK_CORE_APK = "XWalkRuntimeLib.apk";
+
+    private static final String XWALK_CORE_MARKET_URL =
+            "market://details?id=" + XWalkCoreWrapper.XWALK_APK_PACKAGE;
+
+    private static final String XWALK_CORE_DOWNLOAD_URL =
+            "http://10.0.2.2/" + XWALK_CORE_APK; // For test only
```

Comment:
If is test only, can we use more accurate var name like `XWALK_CORE_TESTING_DOWNLOAD_URL`?